oauthenticator/azuread.py

```diff
@@ -39,6 +47,14 @@ def _authorize_url_default(self):
     def _token_url_default(self):
         return f"https://login.microsoftonline.com/{self.tenant_id}/oauth2/token"
 
+    async def update_auth_model(self, auth_model, **kwargs):
+        auth_model = await super().update_auth_model(auth_model, **kwargs)
+
+        user_info = auth_model["auth_state"][self.user_auth_state_key]
+        auth_model["groups"] = user_info.get(self.user_groups_claim)
```

Comment:
The authenticator will always use `name` for the resolved username, so I think it would be consistent to use `groups` instead of `self.user_groups_claim`

Reply:
> Can self.user_groups_claim be something other than groups?

Yes. This depends on how the authentication flow is configured. `groups` and `grp` are both common, however I'm sure there are others.

Reply:
> Let's do this step if the group claim is set, otherwise return the unchanged auth_model.

Awesome, thanks!

Should the `user_info` include an empty list of `groups` if `self.user_groups_claim` is not set?